react/props.md

```diff
@@ -33,228 +32,233 @@
 
-We have explored passing information as props, but we can pass behavior as well. The process is very similar. In this example, we can create a button that when clicked will greet different people from our array. To accomplish this, we can pass both the state object and a method to our child component.
+We can add a method in `App.js` that will use the state setter method named `setPeople` to trigger a `prompt()` method. If we save the prompt to a variable we can capture the result of what our user types. Then it can be added to the array in state.
 
 **src/App.js**
 
 ```javascript
-import React, { Component } from "react";
-import GreetPerson from "./GreetPerson";
-
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      people: ["Bob", "Teddy", "Joe"],
-      currentPerson: 0,
-    };
-  }
-  handleGreeting = () => {
-    let nextPerson = Math.floor(Math.random() * this.state.people.length);
-    this.setState({ currentPerson: nextPerson });
-  };
-
-  render() {
-    return (
-      <GreetPerson
-        person={this.state.people[this.state.currentPerson]}
-        greeting={this.handleGreeting}
-      />
-    );
+import { useState } from "react"
+import Greeter from "./components/Greeter"
+import AddPerson from "./components/AddPerson"
+
+const App = () => {
+  const [people, setPeople] = useState(["Hamilton", "Washington", "Jefferson"])
+
+  const addPerson = () => {
+    const newPerson = prompt()
+    setPeople([...people, newPerson])
   }
+
+  return (
+    <>
+      <h1>Greeter Application</h1>
+      {people.map((person, index) => {
+        return <Greeter name={person} key={index} />
+      })}
+      <AddPerson addPerson={addPerson} />
+    </>
+  )
 }
-export default App;
+
+export default App
 ```
 
-Now we are passing both data and behavior to our child component `GreetPerson`:
+The method we wrote will update the array in state. There are no updates that need to happen to `Greeter` as the code we wrote is dynamic. But the method is not yet being called. We need to create a button with an `onClick` event.
 
-1. Data: the component has access to `this.state.people` as `this.props.person` through the variable `person`
-2. Behavior: the component has access to `handleGreeting` as `this.props.greeting` through the variable `greeting`
+The button's `onClick` will call the method `addPerson` that is being passed as props down into the component.
 
-**src/components/GreetPerson.js**
+**src/components/AddPerson.js**
 
 ```javascript
-import React, { Component } from "react";
-
-class GreetPerson extends Component {
-  render() {
-    return (
-      <div>
-        <h1>Hello, {this.props.person}!</h1>
-        <button onClick={this.props.greeting}>Greet Next Person</button>
-      </div>
-    );
-  }
+const AddPerson = (props) => {
+  return (
+    <>
+      <button onClick={props.addPerson}>Add Person</button>
+    </>
```

Comment:
It seems if our goal is to simply return a single jsx element we'd be served by using the inline return keyword syntax.

```javascript 
return  <button onClick={props.addPerson}>Add Person</button>
```

Reply:
Yeah true. That is a better practice. I will change it.